### PR TITLE
Fixes #19390 and #19391 - using ssl_ca_path prevents using system trusted CA certs for verification

### PR DESCRIPTION
**Pull request by mbacovsky**
Both related commits are in one PR for easier testing (can separate them if it is a problem)

Details about changes are in commit descriptions.

Things worth testing:
- one server, CA in system trust
- one server, CA fetched localy
- two servers, one with CA in system trust, one fetched localy 
- handling of expired local CA cert
- fetching of cert from server that is missing CA cert in the chain

### CODE REVIEW

**Comment by mbacovsky** [comment]
Improved SLL verification error handling is in https://github.com/theforeman/hammer-cli-foreman/pull/308

Server that is missing CA cert in the chain could be simulated by bundling server cert in the ca cert on the server. If the server cert is second in the bundle it gets to the top of the chain and hammer tries to download it (at least with my Apache)

@ares, @tstrachota could you please test/review this?

**Comment by ares** [comment]
I didn't do the full code review but it works exactly as I expected, I didn't test expired fetched cert but all the rest seems to work just fine.

**Comment by mbacovsky** [comment]
@ares Thanks for testing! 
The expired cert should lead to same SSL verification error as with an invalid cert so I used non-CA cert to verify this. The output is copy pasted in the theforeman/hammer-cli-foreman#308. Could you please check the message there (if it makes sense and is helpful)?

**Comment by ares** [comment]
The message looks good to me, few suggestions though:
1) Could the path to the certificate be [expanded](https://ruby-doc.org/core-1.9.3/File.html#method-c-expand_path)? 
2) "Make sure the location contain valid CA certificate for https://sat....com"
* s/contain/contains/
* explicitly saying "contains an unexpired and valid CA certificate"

File: lib/hammer_cli/ssloptions.rb [hunk 1]
```diff
@@ -1,31 +1,40 @@
...
       end
       ssl_options.merge!(cert_key_options)
 
-      ssl_options[:ssl_ca_path] = DEFAULT_SSL_CA_PATH if ssl_options[:ssl_ca_path].nil?
+      # enable ssl verification
+      ssl_options[:verify_ssl] = true if ssl_options[:verify_ssl].nil?
+
+      if ssl_options[:verify_ssl] && uri && !ssl_options[:ssl_ca_file] && !ssl_options[:ssl_ca_path]
+        uri = URI.parse(uri) if uri.is_a?(String)
+        ca_manager = HammerCLI::CACertManager.new(DEFAULT_LOCAL_CA_STORE_PATH)
```
Comment (tstrachota) [inline lib/hammer_cli/ssloptions.rb]: Could you inject cert manager with default value of `HammerCLI::CACertManager.new(DEFAULT_LOCAL_CA_STORE_PATH)`? 
It's nicely isolated functionality so it's a pity to hardcode the instance here.

**Comment by tstrachota** [comment]
One comment in the code, otherwise it looks good

**Comment by mbacovsky** [comment]
Updated.
@tstrachota injection added. I'll squash the new commit into the right commit before merge
@ares thanks for the feedback. it was adderessed in https://github.com/theforeman/hammer-cli-foreman/pull/308

File: config/cli_config.template.yml [hunk 2]
```diff
@@ -51,3 +51,6 @@
 
   # Enable standard authentication in addition to client certificate authentication
   #:ssl_with_basic_auth: true
+
+  # local CA cert store path
+  #:local_ca_store_path: '~/.hammer/certs'
```
Comment (tstrachota) [inline config/cli_config.template.yml]: New line at the end of file, please

**Comment by tstrachota** [comment]
:+1: new line, squash and we can merge!

**Comment by mbacovsky** [comment]
@tstrachota, fixed, squashed. Feel free to merge when the tests are green.

**Pull request merged**